include/mbgl/style/expression/expression.hpp

```diff
@@ -231,52 +231,51 @@ class Expression {
     virtual std::string getOperator() const = 0;
 
 protected:
+    using ExprEquality = decltype(&Expression::operator==);
+    static_assert(std::is_nothrow_invocable_v<ExprEquality, const Expression&, const Expression&>);
```

Comment:
This will always return true.

Reply:
Yes, as long as the method remains declared `noexcept`.  I left it as much as a reminder of the assumption, but I'll remove it if you prefer.